includes/Template_Post_Type.php

```diff
@@ -0,0 +1,93 @@
+ * limitations under the License.
+ */
+
+namespace Google\Web_Stories;
+
+/**
+ * Class Template_Post_Type.
+ */
+class Template_Post_Type {
+	/**
+	 * The slug of the template post type.
+	 *
+	 * @var string
+	 */
+	const POST_TYPE_SLUG = 'web-story-template';
+
+	/**
+	 * Registers the post type for story templates.
+	 *
+	 * @return void
+	 */
+	public static function init() {
+		register_post_type(
+			self::POST_TYPE_SLUG,
+			[
+				'labels'       => [
+					'name'                     => _x( 'Templates', 'post type general name', 'web-stories' ),
+					'singular_name'            => _x( 'Template', 'post type singular name', 'web-stories' ),
+					'add_new'                  => _x( 'New', 'story', 'web-stories' ),
+					'add_new_item'             => __( 'Add New Template', 'web-stories' ),
+					'edit_item'                => __( 'Edit Template', 'web-stories' ),
+					'new_item'                 => __( 'New Template', 'web-stories' ),
+					'view_item'                => __( 'View Template', 'web-stories' ),
+					'view_items'               => __( 'View Templates', 'web-stories' ),
+					'search_items'             => __( 'Search Templates', 'web-stories' ),
+					'not_found'                => __( 'No templates found.', 'web-stories' ),
+					'not_found_in_trash'       => __( 'No templates found in Trash.', 'web-stories' ),
+					'all_items'                => __( 'All Templates', 'web-stories' ),
+					'archives'                 => __( 'Template Archives', 'web-stories' ),
+					'attributes'               => __( 'Template Attributes', 'web-stories' ),
+					'insert_into_item'         => __( 'Insert into template', 'web-stories' ),
+					'uploaded_to_this_item'    => __( 'Uploaded to this story', 'web-stories' ),
+					'featured_image'           => __( 'Featured Image', 'web-stories' ),
+					'set_featured_image'       => __( 'Set featured image', 'web-stories' ),
+					'remove_featured_image'    => __( 'Remove featured image', 'web-stories' ),
+					'use_featured_image'       => __( 'Use as featured image', 'web-stories' ),
+					'filter_items_list'        => __( 'Filter stories list', 'web-stories' ),
+					'items_list_navigation'    => __( 'Templates list navigation', 'web-stories' ),
+					'items_list'               => __( 'Templates list', 'web-stories' ),
+					'item_published'           => __( 'Template published.', 'web-stories' ),
+					'item_published_privately' => __( 'Template published privately.', 'web-stories' ),
+					'item_reverted_to_draft'   => __( 'Template reverted to draft.', 'web-stories' ),
+					'item_scheduled'           => __( 'Template scheduled', 'web-stories' ),
+					'item_updated'             => __( 'Template updated.', 'web-stories' ),
+					'menu_name'                => _x( 'Templates', 'admin menu', 'web-stories' ),
+					'name_admin_bar'           => _x( 'Template', 'add new on admin bar', 'web-stories' ),
+				],
+				'supports'     => [
```

Comment:
Should this be the same as https://github.com/google/web-stories-wp/blob/master/includes/Story_Post_Type.php#L122

```
[
   'title', // Used for amp-story[title].
   'author',
   'editor',
   'excerpt',
   'thumbnail', // Used for poster images.
   'web-stories',
   'revisions', // Without this, the REST API will return 404 for an autosave request.
]

```

Reply:
Templates cannot be edited in the editor, so no need for `editor`.
There's no reason why they'd need revisions/autosaving
And 'web-stories' is just plain wrong there, no idea what that is doing there.